crates/yulgen/src/mappers/assignments.rs

```diff
@@ -20,8 +21,7 @@ pub fn assign(context: &mut FnContext, stmt: &Node<fe::FuncStmt>) -> yul::Statem
         let value = expressions::expr(context, value_node);
 
         let target_attributes = context.expression_attributes(target_node);
-        let typ = FixedSize::try_from(target_attributes.typ.clone()).expect("invalid attributes");
-
+        let typ = target_attributes.typ.as_evm_sized();
```

Comment:
`as_evm_sized()` will panic if a wrong type makes it here but it's been like that before with that `FixedSize::try_from(..).expec(..)` pattern.